smart-contract/assembly/contracts/internals/storageKeys/metadataKeys.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  FILE_METADATA_LOCATION_TAG,
+  FILE_METADATA_TAG,
+  GLOBAL_METADATA_TAG,
+} from './tags';
+
+/**
+ * Generates the storage key for the global metadata.
+ * @param metadataKey - The key of the metadata.
+ * @returns The storage key for the global metadata as a StaticArray<u8>.
+ *
+ * @remarks
+ * Storage representation: [GLOBAL_METADATA_TAG][metadataKey] = metadata_value
+ */
+export function globalMetadataKey(
+  metadataKey: StaticArray<u8>,
+): StaticArray<u8> {
+  return GLOBAL_METADATA_TAG.concat(metadataKey);
+}
+
+/**
+ * Generates the storage key for the file metadata.
+ * @param hashLocation - The hash of the file location.
+ * @param metadataKey - The key of the metadata. Defaults to an empty array if not provided.
+ * @returns The storage key for the file metadata as a StaticArray<u8>.
+ *
+ * @remarks
+ * Storage representation: [FILE_METADATA_TAG][hash(location)][metadataKey] = metadata_value
+ */
+export function fileMetadataKey(
+  hashLocation: StaticArray<u8>,
+  metadataKey: StaticArray<u8> = [],
+): StaticArray<u8> {
+  return FILE_METADATA_TAG.concat(hashLocation).concat(metadataKey);
+}
+
+/**
+ * Generates the storage key for the file metadata location.
+ * @param hashLocation - The hash of the file location.
+ * @returns The storage key for the file metadata location as a StaticArray<u8>.
+ *
+ * @remarks
+ * Storage representation: [FILE_METADATA_TAG][FILE_METADATA_LOCATION_TAG][hash(location)] = metadata_location
```

Comment:
why not represent it as `["file_tag"][hash(location)]["file_metadata_tag"][metadata_entry_key] -> metadata_entry_value` so that everything related to a given file is under the prefix `["file_tag"][hash(file_location)]` ?

See https://forum.massa.community/t/community-feedback-proposal-for-improving-website-storage-on-deweb/82/15

Reply:
The goal was to be able to list all locations without the need to fetch all entries related to all files.